src/transformers/modeling_lxmert.py

```diff
@@ -46,6 +49,96 @@ def __init__(self):
+    (note: the visual encoder in Lxmert is referred to as the "relation-ship" encoder")
+
+
+    Args:
+        language_output (:obj:`torch.FloatTensor` of shape :obj:`(batch_size, sequence_length, hidden_size)`):
+            Sequence of hidden-states at the output of the last layer of the language encoder.
+        vision_output (:obj:`torch.FloatTensor` of shape :obj:`(batch_size, sequence_length, hidden_size)`):
+            Sequence of hidden-states at the output of the last layer of the visual encoder.
+        pooled_output (:obj:`torch.FloatTensor` of shape :obj:`(batch_size, hidden_size)`):
+            Last layer hidden-state of the first token of the sequence (classification, CLS, token)
+            further processed by a Linear layer and a Tanh activation function. The Linear
+        language_hidden_states (:obj:`tuple(torch.FloatTensor)`, `optional`, returned when ``output_hidden_states=True`` is passed or when ``config.output_hidden_states=True``):
+            Tuple of :obj:`torch.FloatTensor` (one for input features + one for the output of each cross-modality layer)
+            of shape :obj:`(batch_size, sequence_length, hidden_size)`.
+        vision_hidden_states (:obj:`tuple(torch.FloatTensor)`, `optional`, returned when ``output_hidden_states=True`` is passed or when ``config.output_hidden_states=True``):
+            Tuple of :obj:`torch.FloatTensor` (one for input features + one for the output of each cross-modality layer)
+            of shape :obj:`(batch_size, sequence_length, hidden_size)`.
+        language_attentions (:obj:`tuple(torch.FloatTensor)`, `optional`, returned when ``output_attentions=True`` is passed or when ``config.output_attentions=True``):
+            Tuple of :obj:`torch.FloatTensor` (one for each layer) of shape
+            :obj:`(batch_size, num_heads, sequence_length, sequence_length)`.
+            Attentions weights after the attention softmax, used to compute the weighted average in the self-attention
+            heads.
+        vision_attentions (:obj:`tuple(torch.FloatTensor)`, `optional`, returned when ``output_attentions=True`` is passed or when ``config.output_attentions=True``):
+            Tuple of :obj:`torch.FloatTensor` (one for each layer) of shape
+            :obj:`(batch_size, num_heads, sequence_length, sequence_length)`.
+            Attentions weights after the attention softmax, used to compute the weighted average in the self-attention
+            heads.
+        cross_encoder_attentions (:obj:`tuple(torch.FloatTensor)`, `optional`, returned when ``output_attentions=True`` is passed or when ``config.output_attentions=True``):
+            Tuple of :obj:`torch.FloatTensor` (one for each layer) of shape
+            :obj:`(batch_size, num_heads, sequence_length, sequence_length)`.
+            Attentions weights after the attention softmax, used to compute the weighted average in the self-attention
+            heads.
+    """
```

Comment:
Several things we have to respect with these model outputs:

- There should be absolutely no different behavior between using the traditional tuple output and this model output, if using `.to_tuple`. This means that:
```py
model(**inputs) == model(**inputs, return_dict=True).to_tuple()
```
We're adding a test to the common tests about this [here](https://github.com/huggingface/transformers/pull/6445).
- Model outputs should therefore have the same order as the regular tuple outputs.
- I've re-ordered it as such: CATEGORY_1[language, vision, misc], CATEGORY_2[language, vision, misc], etc.
- The documented args should have the same order as the args defined in the model output.